src/app/clusters/general_diagnostics_server/general_diagnostics_server.cpp

```diff
@@ -107,26 +108,82 @@ CHIP_ERROR GeneralDiagosticsAttrAccess::Read(const ConcreteAttributePath & aPath
     }
     case TotalOperationalHours::Id: {
-        return ReadIfSupported(&PlatformManager::GetTotalOperationalHours, aEncoder);
+        return ReadIfSupported(&DeviceLayer::PlatformManager::GetTotalOperationalHours, aEncoder);
     }
     case BootReasons::Id: {
-        return ReadIfSupported(&PlatformManager::GetBootReasons, aEncoder);
+        return ReadIfSupported(&DeviceLayer::PlatformManager::GetBootReasons, aEncoder);
     }
     default: {
         break;
     }
     }
     return CHIP_NO_ERROR;
 }
+
+class GeneralDiagnosticDelegate : public DeviceLayer::ConnectivityManagerDelegate, public DeviceLayer::PlatformManagerDelegate
+{
+
+    // Gets called when any network interface on the Node is updated.
+    void OnNetworkInfoChanged() override
+    {
+        ChipLogProgress(Zcl, "GeneralDiagnosticDelegate: OnNetworkInfoChanged");
+
+        for (uint16_t index = 0; index < emberAfEndpointCount(); index++)
+        {
+            if (emberAfEndpointIndexIsEnabled(index))
+            {
+                EndpointId endpointId = emberAfEndpointFromIndex(index);
+                if (endpointId == 0)
+                    continue;
+
+                if (emberAfContainsServer(endpointId, GeneralDiagnostics::Id))
+                {
+                    // If General Diagnostics cluster is implemented on this endpoint
+                    MatterReportingAttributeChangeCallback(endpointId, GeneralDiagnostics::Id,
+                                                           GeneralDiagnostics::Attributes::NetworkInterfaces::Id);
+                }
+            }
+        }
+    }
+
+    // Gets called when the device has been rebooted.
+    void OnDeviceRebooted() override
+    {
+        ChipLogProgress(Zcl, "GeneralDiagnosticDelegate: OnDeviceRebooted");
+
+        for (uint16_t index = 0; index < emberAfEndpointCount(); index++)
+        {
+            if (emberAfEndpointIndexIsEnabled(index))
+            {
+                EndpointId endpointId = emberAfEndpointFromIndex(index);
+
+                if (emberAfContainsServer(endpointId, GeneralDiagnostics::Id))
+                {
+                    // If General Diagnostics cluster is implemented on this endpoint
+                    MatterReportingAttributeChangeCallback(endpointId, GeneralDiagnostics::Id,
+                                                           GeneralDiagnostics::Attributes::RebootCount::Id);
+                    MatterReportingAttributeChangeCallback(endpointId, GeneralDiagnostics::Id,
+                                                           GeneralDiagnostics::Attributes::BootReasons::Id);
+                }
+            }
+        }
+    }
+};
+
+GeneralDiagnosticDelegate gDiagnosticDelegate;
+
 } // anonymous namespace
 
 void MatterGeneralDiagnosticsPluginServerInitCallback()
 {
     registerAttributeAccessOverride(&gAttrAccess);
+
+    PlatformMgr().SetDelegate(&gDiagnosticDelegate);
+    ConnectivityMgr().SetDelegate(&gDiagnosticDelegate);
```

Comment:
Do these delegate registrations survive CHIP stack restart?  They are not documented as to whether they do or not...

Reply:
No, it is registered after each start, why it needs to survive over restart?

Reply:
It depends on the definition of "start", right?  This function depends on being called once per process execution.  IT will not work right if it's called more than once as things stand.  If the CHIP stack is shut down and restarted (e.g. via PlatformManager::Shutdown and then a new InitChipStack, etc), either this function will get called again, in which case the attribute access interface bit will be broken, or it will not be called again, in which case are these delegates going to be set up right?

Reply:
MatterGeneralDiagnosticsPluginServerInitCallback should be called only once during the lifetime of per process execution.  Do we have the scenario we shut down the CHIP stack and restart the CHIP stack without restarting the process? If there is the case, then "registerAttributeAccessOverride(&gAttrAccess);" will also not get called, we are not registering the handle for these platform attributes.

Reply:
> MatterGeneralDiagnosticsPluginServerInitCallback should be called only once during the lifetime of per process execution.

That is the intent, I think...  It happens to not be true in practice right now; see https://github.com/project-chip/connectedhomeip/issues/11352

> Do we have the scenario we shut down the CHIP stack and restart the CHIP stack without restarting the process? 

Yes, absolutely.

> If there is the case, then "registerAttributeAccessOverride(&gAttrAccess);" will also not get called,

Yes, but shutting down the stack does not unregister attr access overrides, so this part is fine.

But in general, we need to decide what the call frequency guarantees are around these callbacks and then make sure that everything in them is operating correctly given those frequency guarantees.  Right now, if we actually change `MatterGeneralDiagnosticsPluginServerInitCallback` to be once-per-process-lifetime this code would break.